securedrop/tests/functional/functional_test.py

```diff
@@ -78,7 +78,17 @@ def _prepare_webdriver(self):
             '\n\n[%s] Running Functional Tests\n' % str(
                 datetime.now()))
         log_file.flush()
-        return firefox_binary.FirefoxBinary(log_file=log_file)
+        if "TRAVIS" in os.environ:
+            # In Travis, the Firefox path isn't in /usr/bin, and Selenium
+            # doesn't find it automatically.
+            firefox_path = "/home/travis/firefox-46.0.1/firefox/firefox"
+            firefox = firefox_binary.FirefoxBinary(firefox_path,
+                                                   log_file=log_file)
+        else:
+            # Selenium's PATH inference is smart enough outside Travis.
+            firefox = firefox_binary.FirefoxBinary(log_file=log_file)
+
+        return firefox
```

Comment:
I also don't love this Travis-specific code path, but I understand why you did this and will not futz. I'll file a followup issue though so on a rainy day in the future someone can workaround this (e.g. by adding the path to the Firefox binary to the PATH in the Travis config).

Reply:
> e.g. by adding the path to the Firefox binary to the PATH in the Travis config

Perplexingly that may not work: the Travis-supplied Firefox binary is _already_ in PATH for the travis user! Selenium uses `which` to find the binary, and I _suspect_ that executing the tasks as root breaks the PATH logic. Will find out while digging through #2216.

A discrete issue is well warranted.